src/modules/claims/claims.service.ts

```diff
@@ -445,29 +443,30 @@ export class ClaimsService {
       expirationTimestamp: claimExpirationTimestamp,
     };
 
-    if (registrationTypes.includes(RegistrationTypes.OnChain)) {
+    if (
+      registrationTypes.includes(RegistrationTypes.OnChain) &&
+      publishOnChain
```

Comment:
This might be incorrect because we might want to issue an `onchainProof` but not actually publish (this is what we do for staking KYC and snapshot credentials for example)

Reply:
Ah, right, I missed that

Reply:
Should add a task to add a test for this case?

Reply:
I would better test this in this PR